### ISSUE TITLE
Export `jax._src.scipy.optimize.line_search.line_search`

### ISSUE THREAD

**Issue by Edenhofer** [google/jax#7363]
Is it sensible to export `jax._src.scipy.optimize.line_search.line_search` maybe even as `jax.scipy.optimize.line_search`? From its syntax it is highly similar to [SciPy's implementation](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.line_search.html) (except of course for the `fprime` part). The two implementations are not equivalent (see [this inline comment](https://github.com/google/jax/blob/f0c7427/jax/_src/scipy/optimize/minimize.py#L66)) and the return types are not the same either but having it exposed would make it easier to develop custom solvers

Note, in a [comment in issue 6320](https://github.com/google/jax/issues/6320#issuecomment-813094689) @shoyer voiced concerns that JAX might be doing something wrong in the line search. In case this is a real concern then the line searcher certainly shouldn't be exposed (and any methods relying on it (?)).

Related links:
* JAX's line search: https://github.com/google/jax/blob/5b3d203/jax/_src/scipy/optimize/line_search.py#L250
* SciPy's line searchers: https://github.com/scipy/scipy/blob/d18bb42/scipy/optimize/linesearch.py
* SciPy's line search documentation: https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.line_search.html?highlight=line_search#scipy.optimize.line_search

**Comment by mattjj** [comment]
Thanks for raising this!

Without looking in great detail, this sounds reasonable to me. We'd need to find someone willing to own the effort though.

@shoyer can you say more about your reservations about the line search implementation? (I'll assign this issue to you until you respond, but then feel free to either keep it or assign it back to me!)

**Comment by shoyer** [comment]
The accuracy concerns may be resolved by https://github.com/google/jax/pull/6785

I think this would _probably_ make more sense in JAXopt (https://github.com/google/jaxopt), where we have a bit more flexibility with regards to function signatures & API choices. We've actually been talking about moving over most of jax.scipy.optimize to jaxopt for this reason.

CC @mblondel

**Comment by Edenhofer** [comment]
Personally I think API compatibility with NumPy and SciPy is a major advantage of JAX. #6785 nicely exemplifies that SciPy is the de-facto standard library to which this implementation shall be compared with and which API is desired (e.g. [this line_searcher.py comment](https://github.com/google/jax/blob/5b3d203/jax/_src/scipy/optimize/line_search.py#L262) is quite telling). Different function signatures would make it more difficult to swap out building blocks and port code between JAX and SciPy.
